components/automate-cli/pkg/verifyserver/models/batchcheck.go

```diff
@@ -78,6 +81,149 @@ type Config struct {
+			c.ExternalPG.PGSuperuserPassword = awsManagedServicesConfig.ManagedRdsSuperuserPassword
+
+			c.ExternalOS.OSDomainName = awsManagedServicesConfig.ManagedOpensearchDomainName
+			c.ExternalOS.OSDomainURL = awsManagedServicesConfig.ManagedOpensearchDomainURL
+
+			// os root-ca might be nil in pre deploy state
+			c.ExternalOS.OSCert = awsManagedServicesConfig.ManagedOpensearchCertificate
+			c.ExternalOS.OSUserPassword = awsManagedServicesConfig.ManagedOpensearchUserPassword
+			c.ExternalOS.OSUsername = awsManagedServicesConfig.ManagedOpensearchUsername
+		}
+	}
+
+	if haConfig.IsExistingInfra() {
+		existingInfraConfig := haConfig.ExistingInfra.Config
+		c.Hardware.AutomateNodeIps = existingInfraConfig.AutomatePrivateIps
+		c.Hardware.ChefInfraServerNodeIps = existingInfraConfig.ChefServerPrivateIps
+		automateConfig := haConfig.Automate.Config
+		if automateConfig.EnableCustomCerts {
+			c.appendCertsByIpToNodeCerts(automateConfig.CertsByIP, automateConfig.RootCA)
+		}
+		chefserverConfig := haConfig.ChefServer.Config
+		if chefserverConfig.EnableCustomCerts {
+			c.appendCertsByIpToNodeCerts(chefserverConfig.CertsByIP)
+		}
+		if !haConfig.IsExternalDb() {
+			c.Hardware.PostgresqlNodeIps = existingInfraConfig.PostgresqlPrivateIps
+			c.Hardware.OpenSearchNodeIps = existingInfraConfig.OpensearchPrivateIps
+			postgresqlConfig := haConfig.Postgresql.Config
+			if postgresqlConfig.EnableCustomCerts {
+				c.appendCertsByIpToNodeCerts(postgresqlConfig.CertsByIP, postgresqlConfig.RootCA)
+			}
+			openSearchConfig := haConfig.Opensearch.Config
+			if openSearchConfig.EnableCustomCerts {
+				c.appendCertsByIpToNodeCerts(openSearchConfig.CertsByIP, openSearchConfig.RootCA, openSearchConfig.AdminKey, openSearchConfig.AdminCert)
+			}
+		}
+
+		if haConfig.IsExternalDb() {
+
+			externalPgConfig := haConfig.External.Database.PostgreSQL
+			c.ExternalPG.PGDbUserName = externalPgConfig.DbuserUsername
+			c.ExternalPG.PGDbUserPassword = externalPgConfig.DbuserPassword
+			c.ExternalPG.PGInstanceURL = externalPgConfig.InstanceURL
+
+			// pg root-ca might be nil in pre deploy state
+			c.ExternalPG.PGRootCert = externalPgConfig.PostgresqlRootCert
+			c.ExternalPG.PGSuperuserName = externalPgConfig.SuperuserUsername
+			c.ExternalPG.PGSuperuserPassword = externalPgConfig.SuperuserPassword
+
+			externalOsConfig := haConfig.External.Database.OpenSearch
+			c.ExternalOS.OSDomainName = externalOsConfig.OpensearchDomainName
+			c.ExternalOS.OSDomainURL = externalOsConfig.OpensearchDomainURL
+
+			// os root-ca might be nil in pre deploy state
+			c.ExternalOS.OSCert = externalOsConfig.OpensearchRootCert
+			c.ExternalOS.OSUserPassword = externalOsConfig.OpensearchUserPassword
+			c.ExternalOS.OSUsername = externalOsConfig.OpensearchUsername
+
+			if haConfig.IsAwsExternalOsConfigured() {
+				c.ExternalOS.OSRoleArn = externalOsConfig.Aws.AwsOsSnapshotRoleArn
+			}
+		}
+	}
+
+	// not available in config
+	c.DeploymentState = ""
```

Comment:
These things are needed for batch check API, please create a tracking ticket to address these missing fields' population

Reply:
Okey, https://chefio.atlassian.net/browse/CHEF-3564